### ISSUE TITLE
Rails/SaveBang matcher too loose

### ISSUE THREAD

**Issue by RKushnir** [bbatsov/rubocop#3435]
## Expected behavior

Cop should match `create`, `update` and `save` calls only on `ActiveRecord` models.
## Actual behavior

Cop matches just any method named `create`, `update` or `save`.
## Steps to reproduce the problem

Add an entry to `.rubocop.yml` to enable the cop

``` yaml
Rails/SaveBang:
  Enabled: true
```

Create a file with the following content

``` ruby
class Foo
  def self.create
  end
end

Foo.create
```

Run rubocop to get `Rails/SaveBang: Use create! instead of create if the return value is not checked.`
## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
```

Not sure if it's actually possible to fix, but I'm getting quite a few false positives so for me this cop is not useful. I wonder, how others deal with it?

**Comment by bbatsov** [comment]
Not sure we can do much about this problem. Generally you can add exclusion comments in such situations, that's the common approach to handle false positives. Another approach would be to limit the files checked by the cop.

**Comment by mikegee** [comment]
Isn't this one of those problems that's impossible to solve in Rubocop, due to static analysis of dynamic language? If so, I'd suggest closing.

If you find the cop valuable and don't tend to write methods that look like ActiveRecord's API, enable this cop, otherwise disable it.

**Comment by Drenmi** [comment]
Agree with @mikegee here.
